File: backend/app/schemas/notification.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, Dict, Any
from app.models.enums import NotificationType


class NotificationBase(BaseModel):
    type: NotificationType
    title: str
    message: str
    related_id: Optional[int] = None
    related_type: Optional[str] = None
    metadata: Optional[Dict[str, Any]] = None


class NotificationCreate(NotificationBase):
    user_id: int


class NotificationResponse(BaseModel):
    id: int
    user_id: int
    type: str
    title: str
    message: str
    read: bool
    read_at: Optional[datetime]
    related_id: Optional[int]
    related_type: Optional[str]
    metadata: Optional[Dict[str, Any]]
    created_at: datetime

    class Config:
        from_attributes = True


class NotificationListResponse(BaseModel):
    items: list[NotificationResponse]
    total: int
    unread_count: int
    skip: int
    limit: int


class NotificationUpdate(BaseModel):
    read: Optional[bool] = None

File: packages/shared/src/schemas/notification.ts
import { z } from "zod";

export const NotificationSchema = z.object({
  id: z.number(),
  user_id: z.number(),
  type: z.string(),
  title: z.string(),
  message: z.string(),
  read: z.boolean(),
  read_at: z.string().datetime().nullable().optional(),
  related_id: z.number().nullable().optional(),
  related_type: z.string().nullable().optional(),
  extra_data: z.record(z.unknown()).nullable().optional(),
  created_at: z.string().datetime(),
});

export const NotificationListResponseSchema = z.object({
  items: z.array(NotificationSchema),
  total: z.number(),
  unread_count: z.number(),
  skip: z.number(),
  limit: z.number(),
});

export type Notification = z.infer<typeof NotificationSchema>;
export type NotificationListResponse = z.infer<typeof NotificationListResponseSchema>;

